### PR TITLE
Add mock.assert_called() to tests where we use mocks

### PR DESCRIPTION
**Pull request by rwrzesien**
Resolves https://github.com/golemfactory/concent/issues/284.

### CODE REVIEW

**Comment by rwrzesien** [comment]
@dybi 
> To have nested with istead few in one

Ok, generally it was one of the main changes, but when I was doing this I have found out that they cannot be broken in any readable form. I will bring them back, but it will take some work.

> add autospec=True to mock.patch (this one I wholeheartedly recommend)

I have read about this one now, and I think that it is not for use for us, as we are creating mocked functions more manually, and controlling the parameters manually.

File: concent_api/core/tests/test_auth_force_payment.py [hunk 1]
```diff
@@ -64,17 +64,26 @@ def test_provider_send_force_payment_and_concent_should_return_it_to_requestor_w
...
+                side_effect=fake_responses
+            ) as get_list_of_payments_mock:
+                response_1 = self.client.post(
+                    reverse('core:send'),
+                    data                                = serialized_force_payment,
+                    content_type                        = 'application/octet-stream',
+                )
+
+        make_force_payment_to_provider_mock.assert_called_once()
+        get_list_of_payments_mock.assert_called()
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply concent_api/core/tests/test_auth_force_payment.py]: :+1:

File: concent_api/core/tests/test_auth_force_payment.py [hunk 2]
```diff
@@ -64,17 +64,26 @@ def test_provider_send_force_payment_and_concent_should_return_it_to_requestor_w
...
-                    response_1 = self.client.post(
-                        reverse('core:send'),
-                        data                                = serialized_force_payment,
-                        content_type                        = 'application/octet-stream',
-                    )
+        with mock.patch(
+            'core.message_handlers.base.make_force_payment_to_provider',
+            side_effect=self._make_force_payment_to_provider
+        ) as make_force_payment_to_provider_mock, freeze_time("2018-02-05 12:00:20"):
+            fake_responses = [mock.Mock(), mock.Mock()]
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply concent_api/core/tests/test_auth_force_payment.py]: :+1:

File: concent_api/core/tests/test_auth_get_task_result.py [hunk 3]
```diff
@@ -121,13 +127,19 @@ def test_requestor_forces_get_task_result_and_concent_immediately_sends_rejectio
...
-            with freeze_time("2017-12-01 11:00:05"):
-                response = self.client.post(
-                    reverse('core:send'),
-                    data=serialized_force_get_task_result,
-                    content_type='application/octet-stream',
-                )
+        with mock.patch(
+            'core.transfer_operations.request_upload_status',
+            side_effect=request_upload_status_false_mock
+        ) as request_upload_status_false_mock_function, freeze_time("2017-12-01 11:00:05"):
```
[the remark this answers is not in the document]
Comment (rwrzesien) [reply concent_api/core/tests/test_auth_get_task_result.py]: :+1: done

**Comment by dybi** [comment]
@rwrzesien, you have done some good works here, so there is no reason not to merge it:)
However, I see adding `autospec` as further improvements (which may be done later, not necessarily by you), which will put us on the safe side in terms of acutual number of parameters passed to mocked function. Thanks to `autospec` we will know when some api changes.

**Pull request merged**